Added code for in situ analysis with Ascent

ghweber opened the pull request:
This commit adds integration of Ascent by the ECP ALPINE project to WarpX. It uses the same "hook" as the SENSEI in situ analysis code and the same parameters in the inputs deck (in_situ.start, in_situ.int) but runs analysis through Ascent.

RemiLehe commented:
Thanks for this PR!
Could you add a short text above introducing what this PR does?
Also, could you add documentation on how to use Ascent with WarpX? See for instance: https://github.com/ECP-WarpX/WarpX/blob/master/Docs/source/visualization/sensei.rst

ghweber commented:
I added a short description. Please let me know if that is fine. Currently this integration is still a bit experimental, but we plan to provide documentation as we test this within the ALPINE team.

RemiLehe commented:
Thanks for adding the description!

Would it make sense for the ALPINE team to do the above-mentioned tests *before* merging this pull request? Could you do these tests from the `Alpine-DAV:ascent-integration` branch, or do you need this to be merged first?

ghweber commented:
What tests? (Sorry, I read through the thread again and didn't see any mentioned, just adding a description and documentation, which we plan to do after some internal tests for which it would be great to have things in the main WarpX repository.)

RemiLehe commented:
Sorry that my language was unclear.
By "tests" I meant the internal tests that you want to do. (i.e. "we plan to provide documentation as we test this within the ALPINE team")
Do you need this to be merged in order to do these internal tests?

ghweber commented:
I have some preliminary instructions that are aimed at our team and that someone else is supposed to refine. It would be easiest if they could do this with the main WarpX repository. Furthermore, this way we can check more easily with, e.g., Lixin how easy those instructions are to follow.

BTW, I believe a recent change may have broken SENSEI integration and I tried to fix this in the commit as well. I think someone renamed "mf" to "mf_avg" when trying avoid code duplication but did not change the name everywhere within the #ifdef.

RemiLehe commented:
OK sounds good. 
Then we can review and merge this PR, and wait for the documentation in a later PR.
@atmyers Would you have time to quickly look at the code?

RemiLehe commented on GNUmakefile:
```diff
@@ -28,6 +28,13 @@ USE_PYTHON_MAIN = FALSE
+
+# The following must be absolute paths
+CONDUIT_DIR ?= /project/projectdirs/alpine/ghweber/ascent/uberenv_libs/spack/opt/spack/cray-cnl9-haswell/gcc-5.3.0/conduit-master-w6ilqv5kfgt2wgbftzpssjj3lttgkqkv
+ASCENT_DIR ?= /project/projectdirs/alpine/ghweber/ascent/uberenv_libs/ascent-install
```
For `CONDUIT_DIR` and `ASCENT_DIR`, could you define them in the file `Make.WarpX` instead of GNUMakefile. 
Also, for external libraries (e.g. HDF5, FFTW, etc.), we usually ask the user to provide the path as an environment variable before typing `make`. Could you use a similar pattern for ASCENT?
(see for instance the variable `FFTW_HOME` in `Make.WarpX`)

RemiLehe commented:
@ghweber Thanks a lot for fixing that SENSEI bug by the way!
I was the one introducing it in the first place!

ghweber commented:
I thought I had added a comment yesterday, but it seems that did not get through. I have moved things to Make.WarpX and also added some preliminary documentation. Please let me know if these changes are ok.

RemiLehe commented:
Looks good to me. Thanks for making these modifications and adding the documentation.
